helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/ClusterModelProvider.java

```diff
@@ -207,10 +408,10 @@ public static ClusterModel generateClusterModelFromCurrentState(
    */
   private static Set<AssignableNode> parseAllNodes(ClusterConfig clusterConfig,
       Map<String, InstanceConfig> instanceConfigMap, Set<String> activeInstances) {
-    return activeInstances.parallelStream().map(
-        instanceName -> new AssignableNode(clusterConfig, instanceConfigMap.get(instanceName),
-            instanceName))
-        .collect(Collectors.toSet());
+    return activeInstances.parallelStream()
+        .filter(instance -> instanceConfigMap.containsKey(instance)).map(
```

Comment:
Why is the containsKey check necessary? If we are doing here, we should mention it in the JavaDoc. Also what if it is false? Should we warn/log?

Reply:
It fails in some tests. I think we have some live instances in the cluster without any config. For those instances, we can just ignore it.
Adding a line to the java doc.